### PR TITLE
float take out of inner loop for speed

### PR DESCRIPTION
**Pull request by DavoudTaghawiNejad**
for big network this speed improvement matters

### CODE REVIEW

File: networkx/generators/directed.py [hunk 1]
```diff
@@ -278,11 +280,11 @@ def _choose_node(G, distribution, delta, psum):
     random.seed(seed)
     number_of_edges = G.number_of_edges()
     while len(G)<n:
-        psum_in = number_of_edges + float(delta_in) * len(G)
-        psum_out = number_of_edges + float(delta_out) * len(G)
+        psum_in = number_of_edges + delta_in) * len(G)
```
Comment (dschult) [inline networkx/generators/directed.py]: extra parens still here

Comment (DavoudTaghawiNejad) [inline networkx/generators/directed.py]: Sorry how embarrassing

**Comment by dschult** [comment]
Thanks!

**Comment by jfinkels** [comment]
A better solution would be `from __future__ import division`, then remove the cast to `float` entirely. It seems that the cast to `float` is introduced so that you get real division in `_choose_node()` when dividing by `psum`, is that correct?

**Comment by DavoudTaghawiNejad** [comment]
that makes sense to me.

**Pull request merged**